### DISCUSSION TITLE
Feature Request: Add maxOptions prop

### DISCUSSION THREAD

**Discussion by nithinpeter** [JedWatson/react-select#4427] [Ideas]
### Prop to restrict the number of results 
We need a way to restrict the number of results rendered on the Select/Async select components. 

##### Use case
This is extremely useful if the number of options is too high and if someone wants to limit the items being displayed to improve performance.

##### API
Add `maxOptions` optional prop for Select/AsyncSelect which takes a number.

**Comment by Chopinsky** [comment]
I think the follow up question would be how to determine what options shall be filtered? I think you may be asking for the incremental loading of the options.

**Comment by nithinpeter** [comment]
Not really! Say for example you have an AsyncSelect component, and your search returns 10,000 results. You need a mechanism to restrict the number of items rendered for performance reasons.

It may not make a lot of sense for `Select`, but I think it is useful to have this prop available for `AsyncSelect`.

**Comment by JalilArfaoui** [comment]
Hi @Chopinsky @nithinpeter 

I needed this too, so I pushed this : https://github.com/JedWatson/react-select/pull/3229

Actually, I need it on a non-async select because sometimes I have > 10 000 options, and for performance reason I want to limit the number of shown options, and let the user type to find more ...

**Comment by The-Code-Monkey** [comment]
Is this gonna be added any time soon?

**Comment by ebonow** [comment]
Greetings, I believe the use case for this would be best suitable as a new prop in `filterOptions`
- The select should only be responsible for the number of options after filtering
- The ordering of the results would likely need to be defined which exists as another Feature Request so it is likely these two would be related.
